Frontend/src/components/TaskList.test.tsx

```diff
@@ -0,0 +1,25 @@
+import { describe, expect, it, vi, Mocked } from 'vitest';
+import TestRenderer from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
+import axios from 'axios';
+import TaskList, { getTasks } from './TaskList';
+
+vi.mock('axios');
+
+describe('TaskList.tsx', () => {
+  describe('getTasks()', () => {
+    it('return an array of tasks', async () => {
+      const mockedAxios = axios as Mocked<typeof axios>;
+      mockedAxios.get.mockResolvedValue({ data: { tasks: [] } });
+      const tasks = await getTasks();
+      expect(tasks).toStrictEqual([]);
+    });
+  });
+
+  describe('taskList.tsx', () => {
```

Comment:
Snapshot tests have their advantages and disadvantages (I recommend googling it!). The problem with them is that they update automatically and many folks, including myself, get bored with them and not check their output. 

Have you tried using [react testing library ](https://testing-library.com/docs/)before? It's a library that checks if a particular element has been rendered in the dom.

Reply:
Big plus on this comment from Justyna!

Reply:
I've done some research and I think I understand the problem. For this particular component, I expect it to change over time as the user uses this app. This defeats the purpose of the snapshot test because I would need to update the snapshot every time a new task is added or task status is changed. 
So instead I wrote a couple tests that check if the `<h1>`'s for each category is rendered, as I would not expect those to change. Am I correct in this train of thought?